app/src/androidTest/java/tech/relaycorp/gateway/background/endpoint/EndpointPreRegistrationServiceTest.kt

```diff
@@ -7,50 +7,69 @@ import android.os.Looper
 import android.os.Message
 import android.os.Messenger
 import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.platform.app.InstrumentationRegistry
 import androidx.test.rule.ServiceTestRule
+import kotlinx.coroutines.runBlocking
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
 import org.junit.Rule
 import org.junit.Test
+import tech.relaycorp.gateway.data.disk.SensitiveStore
+import tech.relaycorp.gateway.domain.LocalConfig
+import tech.relaycorp.relaynet.messages.control.ClientRegistrationAuthorization
+import java.nio.charset.Charset
 
-class NonceRequestServiceTest {
+class EndpointPreRegistrationServiceTest {
 
     @get:Rule
     val serviceRule = ServiceTestRule()
 
     @Test
-    fun requestAndReceiveNonce() {
+    fun requestPreRegistration() = runBlocking {
```

Comment:
I couldn't get this to work with `run BlockingTest` (and `delay()` instead of `Thread.sleep()`). More on this below.